src/main/java/nextstep/subway/entity/group/SectionGroup.java

```diff
@@ -0,0 +1,162 @@
+        return new SectionGroup(sections);
+    }
+
+    public List<Long> getStationsId() {
+
+        return sections.stream()
+            .map(Section::getStationIdList)
+            .flatMap(Collection::stream)
+            .distinct()
+            .collect(Collectors.toList());
+    }
+
+    public Section add(Line line, Station upStation, Station downStation, int distance) {
+
+        final Section ascEndUpSection = getEndUpSection();
+
+        SectionAddAction action = SectionAddActionFactory.make(upStation.getId(),
+            downStation.getId(), ascEndUpSection);
+
+        Optional<Section> currentSection = Optional.of(ascEndUpSection);
+
+        while (existNext(currentSection.get()) && !action.isAdd()) {
+
+            currentSection = findNext(currentSection.get());
+
+            action = SectionAddActionFactory.make(upStation.getId(),
+                downStation.getId(), currentSection.get());
+
+        }
+
+        if (action instanceof AddNone) {
+            throw new IllegalArgumentException("상행역과 하행역 둘 중 하나도 전체구간에 포함되지 않습니다.");
+        }
+
+        return action.add(line, upStation, downStation, distance, currentSection.get(), this);
+    }
+
+    private Section getEndUpSection() {
+        return sections.stream()
+            .filter(Section::isUpEndPointSection)
+            .findFirst()
+            .orElseThrow(
+                () -> new IllegalArgumentException("해당 노선의 상행 종점역이 존재하지 않습니다.")
+            );
+    }
+
+    private Optional<Section> findNext(Section now) {
+
+        return sections.stream()
+            .filter(section -> section.getUpStationId().equals(now.getDownStationId()))
+            .findFirst();
+    }
+
+    private boolean existNext(Section now) {
+        return sections.stream()
+            .anyMatch(section -> section.getUpStationId().equals(now.getDownStationId()));
+    }
+
+    public void delete(long deleteStationId) {
+
+        validateDelete(deleteStationId);
+        sections.remove(getEndDownStation());
+    }
+
+    public Section getEndDownStation() {
+        return sections.stream()
+            .filter(Section::isDownEndPointSection)
+            .findFirst()
+            .orElseThrow(
+                () -> new IllegalArgumentException("해당 노선의 하행종점역이 존재하지 않습니다.")
+            );
+    }
+
+    private void validateDelete(long deleteStationId) {
+
+        validateSizeCanBeDeleted();
+        validateDeleteStationIsEndStation(deleteStationId);
+    }
+
+    private void validateSizeCanBeDeleted() {
+        if (sections.size() <= 1) {
+            throw new IllegalArgumentException("구간이 1개인 경우 삭제할 수 없습니다.");
+        }
+    }
+
+    private void validateDeleteStationIsEndStation(long deleteStationId) {
+        if (!getEndDownStation().isEqualsDownStation(deleteStationId)) {
+            throw new IllegalArgumentException("하행 종점역이 아니면 삭제할 수 없습니다.");
+        }
+    }
+
+    public List<Section> getSections() {
+        return sections;
+    }
```

Comment:
방어적 복사를 위해 불변 컬렉션을 리턴해도 좋을 것 같아요 😃 
```suggestion
    public List<Section> getSections() {
        return Collections.unmodifiableList(sections);
    }
```

Reply:
생각지도 못했는데 이렇게되면 일급컬렉션들을 더 안정적이게 사용할 수 있겠네요 👏🏻